src/java.base/share/classes/sun/security/provider/SHA3.java

```diff
@@ -65,8 +66,15 @@ abstract class SHA3 extends DigestBase {
     };
 
     private final byte suffix;
-    private byte[] state = new byte[WIDTH];
-    private long[] lanes = new long[DM*DM];
+    private long[] state = new long[DM*DM];
+
+    // The following array is allocated to size WIDTH bytes, but we only
+    // ever use the first blockSize bytes it (for bytes <-> long conversions)
+    private byte[] byteState = new byte[WIDTH];
```

Comment:
Since we are storing the state in longs now, this "byte <-> long" conversion can be made through a local variable, right? Is there a reason for having this `byteState` field with size WIDTH bytes?

Reply:
This is interesting: if I use WIDTH (or in blockSize) long arrays in the local level, the performance drops a few per cents. Even more when I only declare the local array in the block in which it is used. However, since we really only need 8 bytes, if I allocate that at the beginning of the function, I don't see that performance drop. So I rewrote the output loop in the function and got rid of the class level declaration.

Reply:
Interesting...
Good that we go with the local 8-byte approach, it improves the readability also.